database/user.go

```diff
@@ -112,6 +112,7 @@ type (
 		// its ID.Hex() form.
 		ID                               primitive.ObjectID `bson:"_id,omitempty" json:"-"`
 		Email                            string             `bson:"email" json:"email"`
+		EmailConfirmed                   bool               `bson:"email_confirmed" json:"emailConfirmed"`
```

Comment:
If `EmailConfirmed` is only true if the corresponding token is empty and vice versa, do we need this field? Can't we just return `EmailConfirmationToken == ""` ?

Reply:
Thanks for the reminder. This is another thing I had to add very quickly and I knew I'd have to follow up on.

I will need to keep the field on the struct but I won't store it in the DB anymore. It will be filled when we're marshalling the user struct into JSON. But we can't avoid having the field unless we want to manually augment the payload each time marshal this into JSON.

If you have other ideas, pleas share.